unlock-app/src/pages/staticdemo.js

```diff
@@ -95,18 +103,32 @@ const Demo = ({ lock, locks }) => {
 Demo.propTypes = {
   lock: PropTypes.string,
   locks: UnlockPropTypes.locks.isRequired,
+  haskey: PropTypes.bool.isRequired,
 }
 
 Demo.defaultProps = {
   lock: '',
 }
 
-export const mapStateToProps = ({ locks, router }) => {
+export const mapStateToProps = ({ locks, keys, router }) => {
   const match = router.location.pathname.match(LOCK_PATH_NAME_REGEXP)
   const lock = match ? match[1] : null
+  let haskey = false
```

Comment:
The logic for `hasKey` is duplicated from what we do in `ShowUnlessUserHasKeyToAnyLock` right? This is probably an indication that something is not right.